src/base/logger.cpp

```diff
@@ -89,6 +89,34 @@ void Logger::addPeer(const QString &ip, const bool blocked, const QString &reaso
     emit newLogPeer(msg);
 }
 
+Log::Msg Logger::getMessage(const int id) const
+{
+    QReadLocker locker(&m_lock);
+
+    const int start = m_msgCounter - m_messages.size();
+    const int end = m_msgCounter - 1;
+    const bool inRange = ((start <= id) && (id <= end));
+
+    if (!inRange)
+        return {-1, -1, Log::CRITICAL, {}};
```

Comment:
I would declare it as static const INVALID in appropriate struct.

Reply:
done